cardinal/server/server_test.go

```diff
@@ -122,8 +117,44 @@ func setTestTimeout(t *testing.T, timeout time.Duration) {
 	}()
 }
 
+func TestHealthEndpoint(t *testing.T) {
+	setTestTimeout(t, 10*time.Second)
+	w := inmem.NewECSWorldForTest(t)
+	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
+	assert.NilError(t, w.RegisterTransactions(sendTx))
+	w.AddSystem(func(world *ecs.World, queue *transaction.TxQueue, _ *log.Logger) error {
+		return nil
+	})
+	assert.NilError(t, w.LoadGameState())
+	txh := makeTestTransactionHandler(t, w, DisableSignatureVerification())
+	resp, err := http.Get("http://localhost:4040/health")
+	assert.NilError(t, err)
+	assert.Equal(t, resp.StatusCode, 200)
+	var healthResponse HealthResponse
+	err = json.NewDecoder(resp.Body).Decode(&healthResponse)
+	assert.NilError(t, err)
+	assert.Assert(t, healthResponse.IsServerRunning)
+	assert.Assert(t, !healthResponse.IsGameLoopRunning)
+	ctx := context.Background()
+	w.StartGameLoop(ctx, time.Tick(1*time.Second), nil)
+	isGameLoopRunning := false
+	for !isGameLoopRunning {
+		time.Sleep(200 * time.Millisecond)
+		resp, err = http.Get("http://localhost:4040/health")
+		assert.NilError(t, err)
+		assert.Equal(t, resp.StatusCode, 200)
+		err = json.NewDecoder(resp.Body).Decode(&healthResponse)
+		assert.Assert(t, healthResponse.IsServerRunning)
+		isGameLoopRunning = healthResponse.IsGameLoopRunning
+	}
+	gameObject := NewGameManager(w, txh.Handler)
+	err = gameObject.Shutdown()
+	assert.NilError(t, err)
+
+}
+
 func TestShutDownViaMethod(t *testing.T) {
-	//setTestTimeout(t, 10*time.Second) // If this test is frozen then it failed to shut down, create failure with panic.
+	setTestTimeout(t, 10*time.Second) // If this test is frozen then it failed to shut down, create failure with panic.
 	w := inmem.NewECSWorldForTest(t)
 	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
```

Comment:
Hmm accidentally merged this, will place this in next PR, Transactions on this test should be removed.